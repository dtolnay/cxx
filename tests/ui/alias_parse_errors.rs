// Rustfmt mangles the extern type alias.
// https://github.com/rust-lang/rustfmt/issues/4159
// ...normally would add #[rustfmt::skip], but that seems to interfere with the error spans.
#[cxx::bridge]
mod ffi {
    extern "C" {
        fn unsupported_foreign_item() {}

        type BadGeneric<T> = Bad;
    }

    extern "Rust" {
        /// Incorrect.
        type Alias = crate::Type;
    }
}

fn main() {}
