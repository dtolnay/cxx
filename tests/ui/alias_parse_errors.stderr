error: unsupported foreign item
 --> $DIR/alias_parse_errors.rs:7:9
  |
7 |         fn unsupported_foreign_item() {}
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aliases with generic parameters are not allowed
 --> $DIR/alias_parse_errors.rs:9:9
  |
9 |         type BadGeneric<T> = Bad;
  |         ^^^^^^^^^^^^^^^^^^

error: type alias in extern "Rust" block is not supported
  --> $DIR/alias_parse_errors.rs:14:9
   |
14 |         type Alias = crate::Type;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^
